pdns/calidns.cc

```diff
@@ -181,34 +186,54 @@ try
 {
   struct sched_param param;
   param.sched_priority=99;
+  int arg_i = 1;
```

Comment:
`argv++; argc--;` after every consumed option simplifies the code and makes `arg_i` obsolete